plugins/tekton/src/utils/pipelineRun-utils.ts

```diff
@@ -0,0 +1,394 @@
+    if (!pipelineRun.status) {
+      return task as PipelineTaskWithStatus;
+    }
+    if (!pipelineRun?.status?.taskRuns && isEmpty(taskRuns)) {
+      return {
+        ...task,
+        status: {
+          reason: getStatusReason(pipelineRun?.status?.conditions?.[0].reason),
+        },
+      } as PipelineTaskWithStatus;
+    }
+    if (
+      (!pipelineRun?.status?.taskRuns ||
+        isEmpty(pipelineRun?.status?.taskRuns)) &&
+      !isEmpty(taskRuns)
+    ) {
+      pipelineRun.status.taskRuns = taskRuns;
+    }
+    let mTask = {
+      ...task,
+      status: get(
+        find(pipelineRun.status.taskRuns, { pipelineTaskName: task.name }),
+        'status',
+      ),
+    } as PipelineTaskWithStatus;
+    // append task duration
+    mTask = appendTaskDuration(mTask);
+    // append task status
+    mTask = appendTaskStatus(mTask);
+    return mTask;
+  });
+};
+
+const pushNodesWithNoDependency = (
+  taskList: PipelineTaskWithStatus[],
+  out: PipelineTask[][],
+) => {
+  taskList?.forEach(task => {
+    if (
+      !taskConditions.hasFromDependency(task) &&
+      !taskConditions.hasRunAfterDependency(task)
+    ) {
+      if (out.length === 0) {
+        out.push([]);
+      }
+      out[0].push(task);
+    }
+  });
+  return out;
+};
+
+const pushNodesWithFromDependency = (
+  taskList: PipelineTaskWithStatus[],
+  out: PipelineTask[][],
+) => {
+  taskList?.forEach(task => {
+    if (
+      !taskConditions.hasRunAfterDependency(task) &&
+      taskConditions.hasFromDependency(task)
+    ) {
+      let flag = out.length - 1;
+      for (let i = 0; i < out.length; i++) {
+        for (const t of out[i]) {
+          if (
+            t.taskRef?.name === task?.resources?.inputs?.[0]?.from?.[0] ||
+            t.name === task?.resources?.inputs?.[0]?.from?.[0]
+          ) {
+            flag = i;
+          }
+        }
+      }
+      const nextToFlag = out?.[flag + 1];
+      if (
+        nextToFlag?.[0]?.resources?.inputs?.[0]?.from?.[0] ===
+        task?.resources?.inputs?.[0]?.from?.[0]
+      ) {
+        nextToFlag?.push(task);
+      } else {
+        out.splice(flag + 1, 0, [task]);
+      }
+    }
+  });
+  return out;
+};
+
+const pushNodesWithRunAfterDependency = (
+  taskList: PipelineTaskWithStatus[],
+  out: PipelineTask[][],
+) => {
+  taskList.forEach(task => {
+    if (taskConditions.hasRunAfterDependency(task)) {
+      let flag = out.length - 1;
+      for (let i = 0; i < out.length; i++) {
+        for (const t of out[i]) {
+          if (
+            t.taskRef?.name === task?.runAfter?.[0] ||
+            t.name === task?.runAfter?.[0]
+          ) {
+            flag = i;
+          }
+        }
+      }
+      const nextToFlag = out?.[flag + 1] ?? null;
+      if (nextToFlag?.[0]?.runAfter?.[0] === task?.runAfter?.[0]) {
+        nextToFlag?.push(task);
+      } else {
+        out.splice(flag + 1, 0, [task]);
+      }
+    }
+  });
+  return out;
+};
+
+export const getPipelineTasks = (
```

Comment:
Based on my previous comment you can remove this function altogether along with all the functions used inside this method.